addons/website_slides/controllers/main.py

```diff
@@ -301,17 +301,39 @@ def create_slide(self, *args, **post):
     # --------------------------------------------------
     @http.route('/slides/embed/<int:slide_id>', type='http', auth='public', website=True, sitemap=False)
     def slides_embed(self, slide_id, page="1", **kw):
+        """
+        Detect if the slide is embedded in a website outside of the Odoo database.
+        The detection is there to differentiate cases where the slide is present on a webpage that is different than
+        the website for which the slide was created
+
+        This is how we decide if the slide is embedded or not:
+        if referer_url is Falsly -> Not embedded (it's the default behavior)
+
+        if there is no website_id (which means the slide is available for all websites) ->
+            is_embedded = domain of referer is equal to ANY of the domain of the odoo DB
+        else (There is a website_id, so only availabled for that website) ->
+            is_embedded = domain of referer is equal to domain of website
+
+        This is important because externally embedding the slide will display Share options inside the slide itself,
+        rather than on the bottom of the page in the Odoo slide module
+
+        In case of uncertainty about the situation, we should assume it is NOT embedded
+        """
         # Note : don't use the 'model' in the route (use 'slide_id'), otherwise if public cannot access the embedded
         # slide, the error will be the website.403 page instead of the one of the website_slides.embed_slide.
         # Do not forget the rendering here will be displayed in the embedded iframe
 
         # determine if it is embedded from external web page
-        referrer_url = request.httprequest.headers.get('Referer', '')
-        base_url = request.env['ir.config_parameter'].sudo().get_param('web.base.url')
-        is_embedded = referrer_url and not bool(base_url in referrer_url) or False
-        # try accessing slide, and display to corresponding template
+        referrer_url = request.httprequest.headers.get('Referer', False)
+        referer_domain = tools.url_domain_extract(referrer_url)
         try:
             slide = request.env['slide.slide'].browse(slide_id)
+            website_id = slide.website_id
+            if not website_id:
+                is_embedded = referer_domain and not any(web_id.domain == referer_domain for web_id in request.env['website'].search([('domain', '!=', '')]))
+            else:
+                is_embedded = referer_domain and website_id.domain != referer_domain
```

Comment:
Reversing the condition would make the test simpler and put the simplest case first. The complex condition should also be formatted to avoid fleeing off of the page.

Furthermore, owing to `url_domain_extract` being pretty much broken for any non-trivial case (e.g. straight won't work for Odoo SaaS / PaaS customers --, everything will be matched to `odoo.com` -- or SLD domains), why doesn't this just check if the referer `endswith` the site's domain, prioritizing by longuest match for the case without a bound website?

Reply:
you should probably use _get_http_domain instead of domain that can be a mix with/without scheme, mix with/without trailing slash, ...

It is only in v15 that you can start to trust the domain, since the write on domain is cleaned by the method `_handle_domain`

Reply:
I changed it to `referer_domain.endswith(website_id._get_http_domain())` (and the other one too), and added a check with a longer referer url